helper/query_parser.js

```diff
@@ -0,0 +1,81 @@
+
+var parser     = require('addressit');
+var extend     = require('extend');
+var get_layers = require('../helper/layers');
+var delim      = ',';
+
+module.exports = function(query) {
+  var parsedAddress0 = {};
```

Comment:
It could be nice to move each of these parsed addresses out into their own function. Presumably each one will be tweaked over time, and having them separated will help keep things clear.